# Macports issue: https://trac.macports.org/ticket/71233
# Upstream issue: https://github.com/Open-Cascade-SAS/OCCT/discussions/48

# StdPrs_BRepFont.cxx: error: cannot initialize a variable of type 'const char *'
#   with an rvalue of type 'unsigned char *'

From: dpasukhi <dpasukhi@opencascade.com>
Date: Tue, 27 Aug 2024 10:33:29 +0000 (+0100)
Subject: 0033808: Coding - FreeType Use unsigned point and contour indexing in `FT_Outline`
X-Git-Url: http://git.dev.opencascade.org/gitweb/?p=occt.git;a=commitdiff_plain;h=7236e83dcc1e7284e66dc61e612154617ef715d6;hp=099e0d25243925da349d43e6e1ee0528763cdabe

0033808: Coding - FreeType Use unsigned point and contour indexing in `FT_Outline`

Changes to auto instead of specific type
---

# Fixed paths here, to make compatible with MacPorts patch command.

diff --git src/StdPrs/StdPrs_BRepFont.cxx.orig src/StdPrs/StdPrs_BRepFont.cxx
index ab2d9b3c9f..cd701879b1 100644
--- src/StdPrs/StdPrs_BRepFont.cxx.orig
+++ src/StdPrs/StdPrs_BRepFont.cxx
@@ -457,7 +457,7 @@ Standard_Boolean StdPrs_BRepFont::renderGlyph (const Standard_Utf32Char theChar,
   for (short aContour = 0, aStartIndex = 0; aContour < anOutline->n_contours; ++aContour)
   {
     const FT_Vector* aPntList = &anOutline->points[aStartIndex];
-    const char* aTags      = &anOutline->tags[aStartIndex];
+    const auto* aTags      = &anOutline->tags[aStartIndex];
     const short anEndIndex = anOutline->contours[aContour];
     const short aPntsNb    = (anEndIndex - aStartIndex) + 1;
     aStartIndex = anEndIndex + 1;
